cmd/external-cert-importer/main.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2015 ISRG.  All rights reserved
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package main
+
+import (
+	"time"
+	"fmt"
+	"encoding/csv"
+	"os"
+	"io"
+	"encoding/hex"
+
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/cactus/go-statsd-client/statsd"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/codegangsta/cli"
+	gorp "github.com/letsencrypt/boulder/Godeps/_workspace/src/gopkg.in/gorp.v1"
+
+	"github.com/letsencrypt/boulder/cmd"
+	"github.com/letsencrypt/boulder/core"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/sa"
+)
+
+var datestamp_format string = "2006-01-02 15:04:05"
+
+
+func addOrUpdateCerts(csvFilename string, dbMap *gorp.DbMap) {
+	file, err := os.Open(csvFilename)
+	cmd.FailOnError(err, "Could not open the file for reading")
+	csvReader := csv.NewReader(file)
+		
+	for {
+		record, err := csvReader.Read()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			fmt.Println("Error:", err)
+			return
+		}
+			
+		notAfter, err := time.Parse(datestamp_format, record[3])
+		spkiBytes, err := hex.DecodeString(record[4])
+		lastUpdated, err := time.Parse(datestamp_format, record[7])
+		certDER, err := hex.DecodeString(record[8])
+				
+		externalCert := core.ExternalCert{
+			SHA1: record[0],
+			Issuer: record[1],
+			Subject: record[2],
+			NotAfter: notAfter,
+			SPKI: spkiBytes,
+			Valid: record[5]=="1",
+			EV: record[6]=="1",
+			LastUpdated: lastUpdated,
+			CertDER: certDER,
+		}
+		
+		var count int64
+		err = dbMap.SelectOne(&count, "SELECT COUNT(*) FROM externalCerts WHERE SHA1=?", record[0])
```

Comment:
FWIW, If we can ensure there's a unique field, it's faster to just let the INSERT fail and discard the err.